### PR TITLE
add simplify consetuitive add pattern in simplify expr pass

### PR DESCRIPTION
**Pull request by crazydemo**
To fuse pattern "conv add add relu" from "conv bias_add bn relu" into "conv add relu".
A "simplifyConsecuitiveAdd" pattern are added in "simplify_expr" pass.
Test case is also added in "test_pass_simplify_expr.py"

### CODE REVIEW

File: tests/python/relay/test_pass_simplify_expr.py [hunk 1]
```diff
@@ -512,5 +512,34 @@ def test_concretize_multiple():
...
+    c_data = np.empty(shape).astype("float32")
+    c1 = relay.const(c_data)
+    c2 = relay.const(c_data)
+
+    def before():
+        x = relay.var("x", shape=(1, 16, 16, 16), dtype="float32")
+        w = relay.var("w", shape=(32, 16, 3, 3), dtype="float32")
+        y = relay.nn.conv2d(x, w, padding=(1, 1))
+        y = relay.add(y, c1)
+        y = relay.add(y, c2)
```
Comment (ZhennanQin) [inline tests/python/relay/test_pass_simplify_expr.py:526]: If const happens on left side, e.g.
```
 y = relay.add(c2, y)
```
Will this simplification work as well?

Comment (crazydemo) [reply tests/python/relay/test_pass_simplify_expr.py:526]: Yes, the simplification works, no matter the const on the right or the left side. I have checked in the test_pass_smiplify_expr.py.

Comment (ZhennanQin) [reply tests/python/relay/test_pass_simplify_expr.py:526]: Would you please add a test to cover this case?

Comment (crazydemo) [reply tests/python/relay/test_pass_simplify_expr.py:526]: test case: const on the left is added in test_pass_smiplify_expr.py

**Comment by ZhennanQin** [comment]
LGTM

**Review by ZhennanQin — APPROVED** [review]

File: src/relay/transforms/simplify_expr.cc [hunk 2]
```diff
@@ -585,6 +585,35 @@ class EliminateIdentityRewrite : public DFPatternRewrite {
   DFPattern const_;
 };
 
+/*! \brief Make two consecuitive add able to be constant_folded. */
+class SimplifyConsecuitiveAdd : public DFPatternRewrite {
```
Comment (ciyongch) [inline src/relay/transforms/simplify_expr.cc]: Typo, `Consecutive`?

File: src/relay/transforms/simplify_expr.cc [hunk 3]
```diff
@@ -585,6 +585,35 @@ class EliminateIdentityRewrite : public DFPatternRewrite {
...
+ public:
+  SimplifyConsecuitiveAdd() {
+    x_ = IsWildcard();
+    const1_ = IsConstant();
+    const2_ = IsConstant();
+    DFPattern add_op_ = IsOp("add");
+    pattern_ = add_op_({add_op_({x_, const1_}), const2_});
```
Comment (ciyongch) [inline src/relay/transforms/simplify_expr.cc]: the pattern will not be selected if any input of the add op is non-const, right?

**Review by ciyongch — APPROVED** [review]
LGTM, check and fix the typo.

File: src/relay/transforms/simplify_expr.cc (hunk 2 above)
Comment (crazydemo) [reply src/relay/transforms/simplify_expr.cc]: fixed.

File: src/relay/transforms/simplify_expr.cc (hunk 3 above)
Comment (crazydemo) [reply src/relay/transforms/simplify_expr.cc]: It only works when two add operators have one same input and two const.

**Pull request merged**